subgraphs/_reference_/src/sdk/protocols/generic/poolSnapshot.ts

```diff
@@ -0,0 +1,156 @@
+import {
+  Pool as PoolSchema,
+  PoolDailySnapshot,
+  PoolHourlySnapshot,
+} from "../../../../generated/schema";
+import { SECONDS_PER_DAY, SECONDS_PER_HOUR } from "../../util/constants";
+import { CustomEventType, getUnixDays, getUnixHours } from "../../util/events";
+
+export class PoolSnapshot {
+  pool: PoolSchema;
+  event: CustomEventType;
+  dayID: i32;
+  hourID: i32;
+
+  constructor(pool: PoolSchema, event: CustomEventType) {
+    this.pool = pool;
+    this.event = event;
+    this.dayID = getUnixDays(event.block);
+    this.hourID = getUnixHours(event.block);
+    this.takeSnapshots();
+  }
+
+  private takeSnapshots(): void {
+    if (!this.isInitialized()) {
+      return;
+    }
+
+    const snapshotDayID =
+      this.pool.lastUpdateTimestamp.toI32() / SECONDS_PER_DAY;
+    const snapshotHourID =
+      this.pool.lastUpdateTimestamp.toI32() / SECONDS_PER_HOUR;
+
+    if (snapshotDayID != this.dayID) {
+      this.takeDailySnapshot(snapshotDayID);
+      this.pool.lastSnapshotDayID = snapshotDayID;
+      this.pool.save();
+    }
+
+    if (snapshotHourID != this.hourID) {
+      this.takeHourlySnapshot(snapshotHourID);
+      this.pool.lastSnapshotHourID = snapshotHourID;
+      this.pool.save();
+    }
+  }
+
+  private isInitialized(): boolean {
+    return !!(
```

Comment:
You don’t need the `!!` since the `&&` already returns a boolean.

(damn, seeing that I did the same on the bridges version, my bad)

Also, you’ll want to use the new fields you’ve added for this purpose? Since `_lastDailySnapshotTimestamp` and `_lastHourlySnapshotTimestamp` are not there

Reply:
Yes! Updated this. All good. I no catch either.